src/rabbit_recovery_terms.erl

```diff
@@ -194,11 +194,29 @@ code_change(_OldVsn, State, _Extra) ->
 %%----------------------------------------------------------------------------
 
 open_table(VHost) ->
+    open_table(VHost, false).
+open_table(VHost, HasFailed) ->
     VHostDir = rabbit_vhost:msg_store_dir_path(VHost),
     File = filename:join(VHostDir, "recovery.dets"),
-    {ok, _} = dets:open_file(VHost, [{file,      File},
-                                     {ram_file,  true},
-                                     {auto_save, infinity}]).
+    try
+        {ok, _} = dets:open_file(VHost, [{file,      File},
+                                        {ram_file,  true},
+                                        {auto_save, infinity}])
+    catch _:_ ->
+        case HasFailed of 
+            false ->
+                rabbit_log:warning("open '~p' failed, delete it and try again.", [File]),
+                rabbit_misc:delete_file(File),
+                open_table(VHost, true);
+            _ ->
+                %% If it has failed, quit and restart the broker 
+                %% depending on the system or other means
+                Wait_time = 5000,
+                rabbit_log:warning("open '~p' failed, quit after ~pms.", [File, Wait_time]),
+                timer:sleep(Wait_time),
+                rabbit_misc:quit(0)    
```

Comment:
I'm afraid this is overly opinionated and a change we won't be able to accept. The whole point of having separate virtual host stores is to avoid shutting down a node when one virtual host is incapable of recovering.

Reply:
Ok， I'll submit the PR again later

Reply:
I understand the intent here and the steps you used to trigger a failure. I'll submit a PR that retries up to N times and credit you for the idea. Thanks.